src/new-interface-poc/types/index.ts

```diff
@@ -0,0 +1,36 @@
+import type { ComputedRef, Ref } from "@vue/reactivity"
+import type { IsArray, IsPlainObject } from "./utils"
+
+export type InputControl<T> = {
+  state: Ref<T | undefined>
+  defaultValue: ComputedRef<T | undefined>
+  dirty: ComputedRef<boolean>
+  clear: () => void
+  reset: () => void
+  updateDefaultValue: (newDefaultValue: T) => void
+  // .. all other input metadata
+}
+
+export type FormControl<T> = {
+  controlsTree: FormNode<T>
```

Comment:
Not blocking at all ofc but do you think it would be possible for the `FormControl` to _be_ a `FormNode` itself? It was like this in the old implementation, and it could be useful sometimes.